kernel/src/scan/mod.rs

```diff
@@ -131,7 +131,20 @@ pub struct ScanResult {
     /// plan to call arrow's `filter_record_batch`, you _need_ to extend this vector to the full
     /// length of the batch or arrow will drop the extra rows
     // TODO(nick) this should be allocated by the engine
-    pub mask: Option<Vec<bool>>,
+    raw_mask: Option<Vec<bool>>,
+}
+
+impl ScanResult {
+    /// Returns the raw mask. This is dangerous to use because it may be shorter than expected.
```

Comment:
nit: I think this comment could be a bit clearer. Something like: "This is dangerous to use because the raw_mask may be shorter than the number of rows returned from a scan".

The extra detail is important because I expect that the docs for `fn raw_mask` will be read more than its field counterpart since that's now the public interface that an LSP is likely to show.

Reply:
Good catch! The original doc comment won't show up at all now, because the field is private.